route/table.go

```diff
@@ -273,8 +273,8 @@ func (t Table) route(host, path string) *Route {
 
 // normalizeHost returns the hostname from the request
 // and removes the default port if present.
-func normalizeHost(req *http.Request) string {
-	host := strings.ToLower(req.Host)
+func normalizeHost(host string, req *http.Request) string {
```

Comment:
The reason I've passed in `*http.Request` was that I needed both host and the TLS state. If you pass in the `host` then you might as well pass in the TLS state, e.g.

```go
func normalizeHost(host string, tls bool) string {
    host = strings.ToLower(host)
    if !tls && strings.HasSuffix(host, ":80") {
        return host[:len(host)-len(":80")]
    }
    if tls && strings.HasSuffix(host, ":443") {
        return host[:len(host)-len(":443")]
    }
    return host
}

...

normalizeHost(req.Host, req.TLS != nil)
```

Reply:
Yup, good call.